models/kernels/EtPriestleyTaylorMethodEx.hpp

```diff
@@ -0,0 +1,263 @@
+//        http://www.fao.org/3/X0490E/x0490e06.htm#aerodynamic%20resistance%20(ra)
+// the zero plane roughness length,"d" can be approximated as 2/3 of the vegetation height (H): d=2/3*H.
+// the momentum roughness height "zom" can be estimated as 0.123*H.
+// the heat transfer roughness height "zoh" can be approximated as 0.1 * zom.
+//-----------------------------------------------------------------------------------------------------------------
+
+double priestley_taylor_method_realization()
+{
+
+// FLAGS
+int yes_aorc; // if TRUE then using AORC forcing data- if FALSE then we must calculate incoming short/longwave rad.
+int yes_wrf;  // if TRUE then we get radiation winds etc. from WRF output.  TODO not implemented.
+
+struct aorc_forcing_data aorc;
+
+struct evapotranspiration_options et_options;
+struct evapotranspiration_params  et_params;
+struct evapotranspiration_forcing et_forcing;
+struct intermediate_vars inter_vars;
+
+struct surface_radiation_params   surf_rad_params;
+struct surface_radiation_forcing  surf_rad_forcing;
+
+struct solar_radiation_options    solar_options; 
+struct solar_radiation_parameters solar_params;
+struct solar_radiation_forcing    solar_forcing;
+struct solar_radiation_results    solar_results;
+
+
+double wind_speed_at_2_m;
+double et_m_per_s;
+double et_mm_per_d;
+double numerator,denominator;
+double saturation_vapor_pressure_Pa;
+double actual_vapor_pressure_Pa;
+
+//###################################################################################################
+// THE VALUE OF THESE FLAGS DETERMINE HOW THIS CODE BEHAVES.  CYCLE THROUGH THESE FOR THE UNIT TEST.
+//###################################################################################################
+// Set this flag to TRUE if meteorological inputs come from AORC
+et_options.yes_aorc = TRUE;                      // if TRUE, it means that we are using AORC data.
+
+// set the et_options method value.  Only one of these should be TRUE
+et_options.use_energy_balance_method   = FALSE;  
+et_options.use_aerodynamic_method      = FALSE;
+et_options.use_combination_method      = FALSE;
+et_options.use_priestley_taylor_method = TRUE;
+et_options.use_penman_monteith_method  = FALSE; 
+
+// -----UNIT TEST RESULTS:-----
+
+// -----Priestley-Taylor method: -----
+// calculated instantaneous potential evapotranspiration (PET) =8.249098e-08 m/s
+// calculated instantaneous potential evapotranspiration (PET) =7.127221 mm/d
+
+
+//###################################################################################################
+// MAKE UP SOME TYPICAL AORC DATA.  THESE VALUES DRIVE THE UNIT TESTS.
+//###################################################################################################
+//read_aorc_data().  TODO: These data come from some aorc reading/parsing function.
+//---------------------------------------------------------------------------------------------------------------
+
+aorc.incoming_longwave_W_per_m2     =  117.1;
```

Comment:
I'm not sure of the purpose of these Ex files, are these just examples?  If so, then hard coding these values is OK, otherwise need to be structured and passed to various ET methods in a way that model formulations can use a consistent calling pattern but explicitly call different methods.